ui/pages/confirmation/ResultTemplate.ts

```diff
@@ -0,0 +1,149 @@
+import { ApprovalType } from '@metamask/controller-utils';
+import {
+  ApprovalRequest,
+  ErrorOptions,
+  SuccessOptions,
+} from '@metamask/approval-controller';
+import { IconName, IconSize } from '../../components/component-library';
+import {
+  FontWeight,
+  BlockSize,
+  AlignItems,
+  FlexDirection,
+  JustifyContent,
+  TypographyVariant,
+  TextAlign,
+  IconColor,
+  BackgroundColor,
+} from '../../helpers/constants/design-system';
+import { resolvePendingApproval } from '../../store/actions';
+import {
+  processError,
+  processHeader,
+  processString,
+  TemplateRendererComponent,
+} from './util';
+
+export type ResultTemplateActions = {
+  resolvePendingApproval: typeof resolvePendingApproval;
+};
+
+export type ResultType = ApprovalType.ResultSuccess | ApprovalType.ResultError;
+
+export class ResultTemplate {
+  #type: ResultType;
+
+  constructor(type: ResultType) {
+    this.#type = type;
+  }
+
+  getValues(
+    pendingApproval: ApprovalRequest<any>,
+    t: (key: string) => string,
+    actions: any,
+  ) {
+    const content = this.#getContent(pendingApproval.requestData || {}, t);
+
+    return {
+      content,
+      submitText: t('ok'),
+      onSubmit: () =>
+        actions.resolvePendingApproval(
+          pendingApproval.id,
+          pendingApproval.requestData,
+        ),
+      networkDisplay: false,
+    };
+  }
+
+  #getContent(
+    requestData: SuccessOptions | ErrorOptions,
+    t: any,
+  ): (undefined | string | TemplateRendererComponent)[] {
```

Comment:
Should we name the type `(undefined | string | TemplateRendererComponent)[]`? It also appears here:

https://github.com/MetaMask/metamask-extension/blob/0c1937854f4141bc03f9b140a46cc8fd901acffd/ui/pages/confirmation/util.ts#L10